coalib/coala_main.py

```diff
@@ -104,27 +103,27 @@ def run_coala(console_printer=None,
         while True:
             for i, action in enumerate(all_actions_possible, 1):
                 console_printer.print(format_lines('{}'.format(
-                    action), symbol='['))
+                    action), symbol='[', line_nr=' ' + str(i-1) + '.'))
 
             line = format_lines(STR_ENTER_NUMBER, symbol='[')
 
             choice = input(line)
 
-            if choice.isalpha():
+            if choice.isalnum():
                 choice = choice.upper()
-                choice = '(' + choice + ')'
-                if choice == '(N)':
+                choice = '(' + choice + ')' if choice.isalpha() else choice
+                if choice == '(N)' or choice == '0':
                     apply_single = 'Do (N)othing'
                     break
                 for i, action in enumerate(all_actions_possible, 1):
-                    if choice in action:
+                    if choice in action or choice in str(i-1):
                         apply_single = action
                         break
+
                 if apply_single:
                     break
-                console_printer.print(format_lines(
-                                    'Please enter a valid letter.',
-                                    symbol='['))
+            console_printer.print(format_lines(
+                'Please enter a valid letter or number.', symbol='['))
```

Comment:
don't change this message for now. Open a different issue, if you feel its not descriptive enough. More newcomer issues that way 😉

Reply:
Okay! I'll keep that in mind and open a issue as soon as this PR gets merged.